R/cluster.R

```diff
@@ -0,0 +1,156 @@
+#' Clustering wrapper
+#'
+#' This function returns a \code{SummarizedExperiment} with clustering 
+#'   information in its colData or rowData
+#'
+#' @param x A
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}
+#'   object.
+#'   
+#' @param data.name A single character value indicating the name of the 
+#'   \code{rowData} (or \code{colData}) where the data will be stored.
+#'   
+#' @param ... Additional parameters to use altExps for example
+#' @inheritParams bluster::clusterRows
+#' @inheritParams runDMN
+#' @inheritParams transformCounts
+#'   
+#'   
+#' @details
+#' This is a wrapper for the \code{clusterRows} function from the 
+#' \link[https://bioconductor.org/packages/release/bioc/html/bluster.html]{bluster} 
+#' package.
+#'
+#' When setting \code{full=TRUE}, the clustering information will be stored in
+#' the metadata of the object.
+#' 
+#' By default, clustering is done on the features.
+#'
+#' @return
+#' \code{cluster} returns an object of the same type as the \code{x} parameter 
+#' with clustering information named \code{clusters} stored in \code{colData} 
+#' or \code{rowData}. 
+#'
+#' @name cluster
+#' @export
+#' 
+#' @author Basil Courbayre
+#'
+#' @examples
+#' library(bluster)
+#' data(GlobalPatterns, package = "mia")
+#' tse <- GlobalPatterns
+#'
+#' # Cluster on rows using Kmeans
+#' tse <- cluster(tse, KmeansParam(centers = 3))
+#' 
+#' # Clustering done on the samples using Hclust
+#' tse <- cluster(tse, 
+#'                MARGIN="samples", 
+#'                HclustParam(metric = "bray", dist.fun = vegan::vegdist))
+#' 
+#' # Getting the clusters
+#' colData(tse)$clusters
+#' 
+NULL
+
+#' @rdname cluster
+#' @export
+setGeneric("cluster", signature = c("x"),
+           function(x, BLUSPARAM, assay.type = assay_name, 
+                    assay_name = "counts", MARGIN="features", full = FALSE, 
+                    name="clusters", data.name="clusters", ...)
+               standardGeneric("cluster"))
+
+#' @rdname cluster
+#' @export
+#' @importFrom bluster clusterRows
+setMethod("cluster", signature = c(x = "SummarizedExperiment"),
+          function(x, BLUSPARAM, assay.type = assay_name, 
+                   assay_name = "counts", MARGIN="features", full = FALSE, 
+                   name="clusters", data.name="clusters", ...) {
+        # Checking parameters
+        MARGIN <- .check_margin(MARGIN)
+        altexp <- .get_altExp(...)
+        se <- .check_and_get_altExp(x, altexp)
+        .check_data_name(se, data.name, MARGIN)
```

Comment:
data.name --> hmmm, how about clust.col?

Reply:
I am unsure to use col or row as it can apply to both